src/argilla/client/feedback/schemas/questions.py

```diff
@@ -147,15 +149,47 @@ def labels_dict_must_be_valid(cls, v: Union[List[str], Dict[str, str]]) -> Union
     @root_validator(skip_on_failure=True)
     def update_settings(cls, values: Dict[str, Any]) -> Dict[str, Any]:
         values["settings"]["type"] = values.get("type")
+
         if isinstance(values.get("labels"), dict):
             values["settings"]["options"] = [
                 {"value": key, "text": value} for key, value in values.get("labels").items()
             ]
-        if isinstance(values.get("labels"), list):
+        elif isinstance(values.get("labels"), list):
             values["settings"]["options"] = [{"value": label, "text": label} for label in values.get("labels")]
-        values["settings"]["visible_options"] = values.get(
-            "visible_labels"
-        )  # `None` is a possible value, which means all labels are visible
+
+        if values.get("visible_labels") == "undefined":
+            if len(values.get("labels", [])) > 20:
+                warnings.warn(
+                    "Since `visible_labels` has not been provided and the total number"
+                    " of labels is greater than 20, `visible_labels` will be set to `20`.",
+                    UserWarning,
+                    stacklevel=1,
+                )
+                visible_labels = 20
+            else:
+                visible_labels = None
+        else:
+            visible_labels = values.get("visible_labels")
+            total_options = len(values.get("labels", []))
```

Comment:
I think we can align the naming to `visible_labels` and call it `total_labels` instead of `total_options`. What do yo think?